meilisearch-http/src/analytics/segment_analytics.rs

```diff
@@ -825,3 +839,53 @@ impl DocumentsDeletionAggregator {
         })
     }
 }
+
+#[derive(Default, Serialize)]
+pub struct HealthAggregator {
+    #[serde(skip)]
+    timestamp: Option<OffsetDateTime>,
+
+    // context
+    #[serde(rename = "user-agent")]
+    user_agents: HashSet<String>,
+
+    total_received: usize,
+}
+
+impl HealthAggregator {
+    pub fn from_query(request: &HttpRequest) -> Self {
+        let mut ret = Self::default();
+        ret.timestamp = Some(OffsetDateTime::now_utc());
+
+        ret.user_agents = extract_user_agents(request).into_iter().collect();
+        ret.total_received = 1;
+        ret
+    }
+
+    /// Aggregate one [DocumentsAggregator] into another.
+    pub fn aggregate(&mut self, other: Self) {
+        if self.timestamp.is_none() {
+            self.timestamp = other.timestamp;
+        }
+
+        // we can't create a union because there is no `into_union` method
+        for user_agent in other.user_agents {
+            self.user_agents.insert(user_agent);
+        }
+        self.total_received = self.total_received.saturating_add(other.total_received);
```

Comment:
You never reset the total received? Is the total received reset after each reboot of the engine?

Reply:
It's resetted when we send the batch

https://github.com/meilisearch/meilisearch/pull/3065/files#diff-49280ed88af4d3056a5b9928d51418c085261c5864d83591d71333a8bcaee158R419-R420